frontend/packages/console-app/src/components/quick-starts/utils/quick-start-context.ts

```diff
@@ -0,0 +1,116 @@
+import { QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY } from '../../../redux/reducers/quick-start-reducer';
+import { createContext, useCallback } from 'react';
+import { QuickStartStatus, QuickStartTaskStatus } from './quick-start-types';
+import { useUserSettings } from '@console/shared/src/hooks/useUserSettings';
+
+export type QuickStartContextValues = {
+  activeQuickStartID: string;
+  allQuickStartStates?: any;
+  activeQuickStartState?: any;
+  setActiveQuickStart?: (quickStartId: string, totalTasks?: number) => void;
+  setQuickStartStatus?: (quickStartId: string, status: QuickStartStatus) => void;
+  setQuickStartTaskNumber?: (quickStartId: string, taskNumber: number) => void;
+  setQuickStartTaskStatus?: (taskStatus: QuickStartTaskStatus) => void;
+  resetQuickStart?: (quickStartId: string, totalTasks: number) => void;
+  getQuickStartForId?: (id: string) => any;
+};
+
+export const QuickStartContext = createContext<QuickStartContextValues>({ activeQuickStartID: '' });
+
+const getInitialState = () =>
+  localStorage.getItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY)
+    ? JSON.parse(localStorage.getItem(QUICK_START_REDUX_STATE_LOCAL_STORAGE_KEY))
+    : {};
+
+const getDefaultQuickStartState = (totalTasks: number) => {
+  const defaultQuickStartState = {
+    status: QuickStartStatus.NOT_STARTED,
+    taskNumber: -1,
+  };
+  if (totalTasks) {
+    for (let i = 0; i < totalTasks; i++) {
+      defaultQuickStartState[`taskStatus${i}`] = QuickStartTaskStatus.INIT;
+    }
+  }
+  return defaultQuickStartState;
+};
+
+const QUICK_START_KEY = 'console.quickstart';
+const ACTIVE_QUICK_START_ID_KEY = `${QUICK_START_KEY}.active`;
+const ALL_QUICK_START_STATE_KEY = `${QUICK_START_KEY}.allStates`;
```

Comment:
nit: will it make sense to add these in const file for `quick-start`

Reply:
They are not being used in any other place. So it's fine to keep it here.